### PR TITLE
WIP: Issue #468 deletion

### PR DESCRIPTION
**Pull request by cesy**
Draft, working towards fixing issue #468. Please do not yet merge, just leaving it here for easy reference and any early review of the test cases.

### CODE REVIEW

File: spec/features/members/member_deletion_spec.rb [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      FactoryGirl.create(:comment, :author => other_member, :post => memberpost)
+    end
+
+    scenario "has option to delete on member profile page" do
+      visit member_path(member)
+      expect(page).to have_link "Delete account"
+    end
+    
+    scenario "requests confirmation for deletion"
+    # need to figure out how to test javascript if we want to bother
```
Comment (pozorvlak) [inline spec/features/members/member_deletion_spec.rb]: Testing JavaScript is no longer a problem - PhantomJS (which we're using for feature tests) embeds a full JavaScript interpreter.

Comment (tygriffin) [inline spec/features/members/member_deletion_spec.rb]: Just be sure to use :js => true to make turn on Phantom.

iPhone$B$+$iAw?.(B

2015/01/21 2:07$B!"(Bpozorvlak <notifications@github.com> $B$N%a%C%;!<%8(B:

**Comment by Skud** [comment]
Just wanted to let you know I've reviewed this up to this point and it's looking good :)

**Comment by Skud** [comment]
Hi @cesy, just wanted to check in on the status of this?  We had tentatively included it in Release 8 (due to happen by the end of this week) but if we do want to get it into that release we'd need to merge a PR within the next day or so.  If you're actively working on this and would like to try for that, please go ahead!  If you don't think you'll make that deadline, please edit the issue (#468) and put it in a later release.

**Comment by cesy** [comment]
Yeah, I'm not going to make Release 8. I'll do my best for Release 9 - work suddenly got busy again.

**Comment by Skud** [comment]
Hi Cesy!

As a result of starting to send more email (the planting reminder emails) I'm getting occasional requests to support to delete people's accounts, from people who signed up but presumably don't use them.  I'm wondering what the status of this story is, as we have an increasingly urgent need for it.

A.

**Comment by cesy** [comment]
Everything I've done is on here, so someone else can take over if you need
it quickly. I can't get the crop bit to work at the moment, and am poking
at it occasionally.

**Comment by cesy** [comment]
I've added a start on the post and comment stuff, and oddly it works when I test manually but not in the automated tests.

**Comment by oshiho3** [comment]
Hi @cesy, I managed to run the scenario "leaves crops behind, reassigned to ex_wrangler" successfully by changing the spec as below.

* The logic we want to run is in the MembersController, so changed the below line from

>      member.delete

to

>      logout
>      login_as(member)
>      visit member_path(member)
>      click_link 'Delete account'
>      logout

* The ex_wrangler member which is being searched in MembersController destroy method needs to exist when the test is run. So use <code>let!</code> to force create the member instead of lazy loading it which won't happen in this case as ex_wrangler isn't referenced in the test scenario.

**Comment by oshiho3** [comment]
Same for "member deletion with activity and followers deletes and..." scenario, replace <code>member.delete</code> in background block with login, click_link Delete account, logout, login by another member...

**Comment by cesy** [comment]
Brilliant, thank you.

**Comment by cesy** [comment]
Okay, I think we finally have functionality that will let a non-admin be deleted safely if they haven't got any orders/payments. We still need the bits to preserve order history, and to require password confirmation.

**Comment by oshiho3** [comment]
You're welcome, @cesy!

**Comment by cesy** [comment]
Having messed up my branches with Nitrous.io being evil, how do I now reopen this?

**Pull request closed without merging**